### PR TITLE
Add atomization-consistent training

### PR DESCRIPTION
**Pull request by lubbersnick**
This PR adds a new feature to hippynn which trains an energy predictor which is self-consistent under an atomization process; the training must be done to an appropriately normalized dataset, that is one of calculated atomization energies.

The feature should be functional with all other hippynn facilities including the ase interface and lammps interface. If it is not working, and you are somehow at this PR because it is broken, please leave a message.

Changes for this feature include:
- addition of atomization consistent node/torch module as well as vacuum feature transformation on the graph level.
- added atomization consistent option to the ani1x_training example
- tests which verify that the atomization energy module can be converted to a normal HEnergy module and used with ASE to produce the same answers.
- changes to the baseline subtraction in ani1x_training example to produce a correct atomization energy using the appropriate vacuum spin states for carbon, nitrogen, and oxygen.

Other related changes introduced here:
- better automatic determination of relative nodes using an ancestor fallback scheme
- allow valuenodes to take on pure python values (e.g. Int) instead of only torch tensors, using a keyword.
- slightly improved imports to base package
- bugfixes to hierarchical initialization and database features.
- improved error message for hip-nn creation when hyperparameters are not fully specified.

### CODE REVIEW

**Comment by lubbersnick** [comment]
Now partially addresses #113

**Comment by lubbersnick** [comment]
also adds fix for #121
